src/mem/ruby/protocol/GPU_VIPER-TCC.sm

```diff
@@ -606,6 +606,7 @@ machine(MachineType:TCC, "TCC Cache")
           out_msg.Destination.add(in_msg.Requestor);
           out_msg.Sender := machineID;
           out_msg.MessageSize := MessageSizeType:Response_Data;
+          out_msg.instSeqNum := in_msg.instSeqNum;
```

Comment:
Is this needed for this commit?  It doesn't seem to affect the wada part.

Reply:
This helped me debug the issue and might be useful in fixing future issues. Without this, I was seeing a sequence number of 0 in all the messages exiting the TCC and it made it hard to track requests.

Reply:
So ultimately the issue is: we want commits to be self-contained, and only contain what is needed to fix a specific bug or add a specific feature, etc.  In this case, it seems like you've combined two things: one to add more information and one to fix a bug.  I understand they are related, but if this commit were rolled back/reverted in the future, then both would be lost.  So I think the right solution would be to have two separate commits.

Reply:
I've split it across two commits and updated the pull request